server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/task/DataTransferTask.java

```diff
@@ -0,0 +1,391 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Builder
+@Slf4j
+public class DataTransferTask implements Callable<DataTransferTaskResult> {
+    private static final Set<String> OUTPUT_FILTER_FILES = new HashSet<>();
+
+    private final DataMaskingService maskingService;
+    private final Holder<DataTransferCallable> jobHolder;
+    private final User creator;
+    private final DataTransferConfig config;
+    private final DataTransferAdapter adapter;
+    private final DataTransferProperties properties;
+    private final File workingDir;
+    private final File logDir;
+    private final ConnectionConfig connectionConfig;
+
+    @Override
+    public DataTransferTaskResult call() throws Exception {
+        try {
+            SecurityContextUtils.setCurrentUser(creator);
+
+            List<URL> inputs = new ArrayList<>();
+            preHandle(inputs);
+
+            DataTransferCallable job = TaskPluginUtil
+                    .getDataTransferExtension(config.getConnectionInfo().getConnectType().getDialectType())
+                    .generate(config, workingDir, logDir, inputs);
+            jobHolder.setValue(job);
+
+            DataTransferTaskResult result = job.call();
+
+            validateSuccessful(result);
+
+            postHandle(result);
+
+            return result;
+
+        } catch (Exception e) {
+            log.warn("Failed to run data transfer task.", e);
+            throw e;
+
+        } finally {
+            SecurityContextUtils.clear();
+        }
+    }
+
+    private void preHandle(List<URL> inputs) throws Exception {
+        config.setUsePrepStmts(properties.isUseServerPrepStmts());
+        if (config.getTransferType() == DataTransferType.IMPORT) {
+            preHandleForImport(inputs);
+        } else {
+            preHandleForExport();
+        }
+    }
+
+    private void postHandle(DataTransferTaskResult result) throws Exception {
+        if (config.getTransferType() == DataTransferType.IMPORT) {
+            postHandleForImport();
+        } else {
+            postHandleForExport(result);
+        }
+    }
+
+    private void validateSuccessful(DataTransferTaskResult result) {
+        List<String> failedObjects = ListUtils.union(result.getDataObjectsInfo(), result.getSchemaObjectsInfo())
+                .stream()
+                .filter(objectStatus -> objectStatus.getStatus() != Status.SUCCESS)
+                .map(ObjectResult::getSummary)
+                .collect(Collectors.toList());
+        Verify.verify(CollectionUtils.isEmpty(failedObjects),
+                "Data transfer task completed with unfinished objects! Details : " + failedObjects);
+    }
+
+    private void preHandleForImport(List<URL> inputs) throws IOException {
+        /*
+         * move import files
+         */
+        List<String> importFileNames = config.getImportFileName();
+        if (config.isCompressed()) {
+            DumperOutput dumperOutput = copyImportZip(importFileNames, workingDir);
```

Comment:
dumperOuput is only for oceanbase which is not general, we should not use it in service layer